src/rosdep2/lookup.py

```diff
@@ -103,7 +103,11 @@ def get_rule_for_platform(self, os_name, os_version, installer_keys, default_ins
         if type(data) != dict:
             raise InvalidData('rosdep value for [%s] must be a dictionary' % (self.rosdep_key), origin=self.origin)
         if os_name not in data:
-            raise ResolutionError(rosdep_key, data, os_name, os_version, 'No definition of [%s] for OS [%s]' % (rosdep_key, os_name))
+            if '*' not in data:
+                raise ResolutionError(rosdep_key, data, os_name, os_version, 'No definition of [%s] for OS [%s]' % (rosdep_key, os_name))
+            elif type(data['*']) != dict:
+                raise InvalidData('rosdep value under OS wildcard for [%s] must specify a package manager' % (rosdep_key))
+            os_name = '*'
```

Comment:
This is a little odd to set the query os_name to `*`. I see it all propagates through. However it will also propagate to the debug statements etc. I think it's slightly more complex but we should bookkeep the queried os and the evaluated os. Otherwise people will set an OS explicitly and then get errors saying that it can't resolve for `*` OS.

Reply:
I'm actually really glad you brought this up. I did this when I originally made the change, but opted for a simpler change to make the review easier. I'll preserve the queried OS name and version as suggested.

Reply:
91cf3a1